### PR TITLE
Merging latest openj9 into openj9-staging - Update OPENJDK_TAG to jdk8u292-b01

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Resolved merging conflicts:

```
  jdk/src/solaris/bin/aarch64/jvm.cfg
  jdk/src/share/native/com/sun/media/sound/SoundDefs.h
  jdk/make/lib/SoundLibraries.gmk
  common/autoconf/generated-configure.sh
```

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
I note there are some changes we could make to help avoid future merge conflicts, although it doesn't need to occur in this PR.

**Comment by pshipton** [comment]
lgtm

**Comment by keithc-ca** [comment]
The title says this merges from `jdk8u292-b00`, but that tag refers to content that is already in the openj9 branch.
Is the intent here to merge with `jdk8u292-b01`?

**Comment by JasonFengJ9** [comment]
> but that tag refers to content that is already in the openj9 branch.
Is the intent here to merge with jdk8u292-b01?

It intended to merge with `jdk8u292-b00`.
What content tagged is already in the `openj9` branch?

**Comment by keithc-ca** [comment]
The tag 'jdk8u292-b00' points to de7aa62cfa02b70641093496899e1abe4e7b05a1 which is an ancestor commit of a1b8db283e3775be94350088a00b8df072eb26bb which is where the openj9 is at currently.

File: closed/openjdk-tag.gmk [hunk 1]
```diff
@@ -1 +1 @@
-OPENJDK_TAG := jdk8u282-b08
+OPENJDK_TAG := jdk8u292-b00
```
Comment (keithc-ca) [inline closed/openjdk-tag.gmk:1]: This should be `jdk8u292-b01`.

File: common/autoconf/jdk-options.m4 [hunk 2]
```diff
@@ -165,9 +165,6 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_JVM_VARIANTS],
   if test "x$VAR_CPU" = xppc64 -o "x$VAR_CPU" = xppc64le ; then
     INCLUDE_SA=false
   fi
-  if test "x$OPENJDK_TARGET_CPU" = xaarch64; then
-    INCLUDE_SA=false
-  fi
```
Comment (keithc-ca) [inline common/autoconf/jdk-options.m4:170]: I don't think we want this change.

File: jdk/make/lib/SoundLibraries.gmk [hunk 3]
```diff
@@ -152,9 +152,8 @@ else
   endif
 
   ifeq ($(OPENJDK_TARGET_CPU), aarch64)
-    LIBJSOUND_CFLAGS += -DX_ARCH=X_AARCH64
+	LIBJSOUND_CFLAGS += -DX_ARCH=X_AARCH64
```
Comment (keithc-ca) [inline jdk/make/lib/SoundLibraries.gmk:155]: We don't really want this change: the leading tab could be problematic.

**Review by keithc-ca — CHANGES_REQUESTED** [review]
I'd like to see SoundDefs.h change so `X_AARCH64` is matches openjdk.

File: common/autoconf/jdk-options.m4 (hunk 2 above)
Comment (JasonFengJ9) [reply common/autoconf/jdk-options.m4:170]: This is from OpenJDK, is there an issue/pr that requires the code snippet being deleted?

File: jdk/make/lib/SoundLibraries.gmk (hunk 3 above)
Comment (JasonFengJ9) [reply jdk/make/lib/SoundLibraries.gmk:155]: sure, we can keep initial JCL patch instead.

File: closed/openjdk-tag.gmk (hunk 1 above)
Comment (JasonFengJ9) [reply closed/openjdk-tag.gmk:1]: agreed

File: common/autoconf/jdk-options.m4 (hunk 2 above)
Comment (keithc-ca) [reply common/autoconf/jdk-options.m4:170]: The change is to delete those lines. We don't want that; we want to continue to support aarch64, so those lines should stay.

Comment (JasonFengJ9) [reply common/autoconf/jdk-options.m4:170]: As per https://bugs.openjdk.java.net/browse/JDK-8155627, this change is to _Enable SA on AArch64_ 
        _The serviceability agent is currently disabled. There is no good reason for this._
I think OpenJ9 can use this change to enable SA.

Comment (keithc-ca) [reply common/autoconf/jdk-options.m4:170]: If we think SA should be enabled, it should be done in a separate pull request specifically for that purpose.

Comment (JasonFengJ9) [reply common/autoconf/jdk-options.m4:170]: This commit comes in with OpenJDK updates as part of merging process, if a change to this OpenJDK commit (i.e., JCL patch) is required, it should be in a separated PR instead, does this make sense?

Comment (keithc-ca) [reply common/autoconf/jdk-options.m4:170]: I misunderstood the source of those lines. They originated with openjdk and were not changed for openj9. We should follow openjdk and remove them too. Sorry for the diversion.

**Comment by JasonFengJ9** [comment]
Updated the PR, please have another look.

**Comment by keithc-ca** [comment]
`jdk/make/closed/autoconf/generated-configure.sh` needs to be regenerated to reflect changes to `common/autoconf/jdk-options.m4`.

**Comment by JasonFengJ9** [comment]
> jdk/make/closed/autoconf/generated-configure.sh needs to be regenerated to reflect changes to common/autoconf/jdk-options.m4.

Re-generated `generated-configure.sh` via `common/autoconf/autogen.sh` https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/476/commits/8cb1be3c5035acc3ffc905bbc976a9e79c1cd615.

**Comment by keithc-ca** [comment]
Please redo `common/autoconf/autogen.sh` and include both generated files. They each have an embedded timestamp which I believe need to be consistent.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins test sanity aix,osx,plinux,win32 jdk8

**Comment by pshipton** [comment]
The Windows failure is https://github.com/eclipse/openj9/pull/11869

**Comment by pshipton** [comment]
Note this could just be merged, as it will through the acceptance build anyway. The only downside is possibly having to open another PR to fix problems.

**Comment by keithc-ca** [comment]
> Note this could just be merged, as it will go through the acceptance build anyway.

That seems reasonable. Please approve if you're happy and I will merge.

**Review by pshipton — APPROVED** [review]

**Pull request merged**